great_expectations/checkpoint/actions.py

```diff
@@ -85,23 +86,42 @@ def _build_renderer(config: dict) -> Renderer:
     return renderer
 
 
+class ValidationActionRunStatus(enum.Enum):
+    SUCCESS = "success"
+    FAILURE = "failure"
+    NOT_RUN = "not_run"
+
+
+class ValidationActionResult(BaseModel):
+    """
+    The result of running a ValidationAction.
+
+    Args:
+        status: The status of the action run. If the action was not run, the status is "not_run".
+        run_info: A dictionary containing information about the run.
+    """
+
+    status: ValidationActionRunStatus
+    run_info: dict
```

Comment:
Does the run_info vary from action to action or can we be more concrete about this?

Reply:
Ah yeah I don't love our current dicts but I'm trying to minimize changes. They're all pretty much `{"action_type_in_snake_case": "success_or_failure"}` so they could be simplified with a simple bool but I think I want to keep it this way for a few reasons:
1. Minimize changes (for now)
2. Allow for more complex `run_info` (`UpdateDataDocs` leverages this and future custom actions might want to include more details?)